File: ContainerWithMostWater.cpp
class Solution
{
public:
    int maxArea(vector<int> &height)
    {
        int n = height.size();
        int maximum = 0;
        int a = 0, b = n - 1;

        int area;
        while (a != b)
        {
            area = (b - a) * min(height[a], height[b]);

            if (area > maximum)
                maximum = area;

            if (height[a] < height[b])
                a++;
            else
                b--;
        }

        return maximum;
    }
};

File: ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0 
        right = len(height) - 1
        maxWater = 0

        while left < right:
            con = min(height[left],height[right])
            maxWater = max(maxWater, con * (right - left))

            if height[left] > height[right]:
                right -= 1
            else:
                left += 1

        return maxWater
